android_mobile/src/main/java/com/alexstyl/specialdates/ReroutingActivity.kt

```diff
@@ -4,36 +4,63 @@ import android.app.Activity
 import android.app.NotificationChannel
 import android.app.NotificationManager
 import android.app.PendingIntent
+import android.content.ComponentName
 import android.content.Context
 import android.content.Intent
 import android.graphics.Color
 import android.os.Bundle
 import android.support.v4.app.NotificationCompat
 import com.alexstyl.android.Version
 import com.alexstyl.specialdates.home.HomeActivity
-import java.time.Clock
+import javax.inject.Inject
 
 class ReroutingActivity : Activity() {
 
+
+    lateinit var errorTracker: CrashAndErrorTracker
+        @Inject set
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
+        (application as MementoApplication).applicationModule.inject(this)
+
         if (BuildConfig.DEBUG) {
             showDebugNotification()
         }
 
-        val intent = HomeActivity.getStartIntent(this)
-        startActivity(intent)
+        val targetIntent: Intent = createRedirectFor(intent)
+
+        startActivity(targetIntent)
 
         finish()
     }
 
+    private fun createRedirectFor(intent: Intent): Intent {
+        return if (intent.isARedirect()) {
+            val rerouteIntent = Intent(intent)
+            val targetActivity = intent.extras.getString(FCM_EXTRA__TARGET_ACTIVITY)
+            rerouteIntent.component = ComponentName(BuildConfig.APPLICATION_ID, targetActivity)
```

Comment:
I think it would be preferable to use a URI schema instead of having the activity name in the intent. If tomorrow you rename an activity you will have a problem with maintaining this (what if half of your user base uses the old app with the old named activity?)

Reply:
Agreed. i'll change this to an implicit call instead (passing an ACTION)